src/costcontextmenu.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+  costcontextmenu.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "costcontextmenu.h"
+
+#include <QTreeView>
+#include <QHeaderView>
+#include <QMenu>
+
+CostContextMenu *CostContextMenu::instance()
+{
+    static CostContextMenu menu;
```

Comment:
static auto menu = QPointer<CostContextMenu>(new CostContextMenu(QCoreApplication::instance());

to ensure this gets reaped at shutdown

better yet: don't use singletons!

Reply:
Is there a alternative to the singleton in this case? The only alternative I found was creating the menu in MainWindow and then passing a reference to all Result*Page and then to the views.

Reply:
yes exactly, pass things around. I mean hotspot is supposed to support multiple main windows but just testing it now I realize this totally broke when I migrated to KDDW :D will have to fix that.

but yes, please create it centrally and then pass it along, otherwise multiple main windows would be synced which is imo not really desired, is it?

Reply:
ok, I added it to ResultsPage